test/lib/tasks/oauth_access_records_test.rb

```diff
@@ -23,4 +23,28 @@ class OauthAccessRecordsTaskTest < ActiveSupport::TestCase
 
     assert_equal 1, user.authorisations.count
   end
+
+  context "deleting grant records" do
+    should "create a single account activity log entry per User" do
+      user = create(:user)
+      create(:access_grant, resource_owner_id: user.id, expires_in: 0)
+      create(:access_grant, resource_owner_id: user.id, expires_in: 0)
+
+      @task.invoke
+
+      assert_equal 1, user.event_logs(EventLog::ACCESS_GRANTS_DELETED).count
+    end
+  end
+
+  context "deleting access records" do
+    should "create a single account activity log entry per User" do
+      user = create(:user)
+      create(:access_token, resource_owner_id: user.id, expires_in: 0)
+      create(:access_token, resource_owner_id: user.id, expires_in: 0)
+
+      @task.invoke
+
+      assert_equal 1, user.event_logs(EventLog::ACCESS_TOKENS_DELETED).count
+    end
+  end
```

Comment:
I'm a bit nervous that these two tests don't provide enough coverage to give us confidence in the implementation. For example, if I remove the `.uniq` from the following line, neither of these tests fail:

```ruby
        all_users = User.where(id: records.map(&:resource_owner_id).uniq)
```

Reply:
It turns out that my use of `#uniq` here was a red herring :fishing_pole_and_fish: `ActiveRecord` appears to do the right thing here without it.

Any suggestions for other gaps in these tests? I can definitely add more coverage, but I'm struggling to think of what's missing and would be valuable

Reply:
> It turns out that my use of `#uniq` here was a red herring 🎣 `ActiveRecord` appears to do the right thing here without it.

Oh, cool! 👍 

> Any suggestions for other gaps in these tests? I can definitely add more coverage, but I'm struggling to think of what's missing and would be valuable

Um 🤔.  Perhap include more than one user in at least one test to ensure we're recording the right events against the right user? Perhaps include at least one assertion that we're recording the events for the right applications? I'm not _sure_ either of those make sense, but just a couple of ideas off the top of my head...

My original comment was triggered by a gut feeling that the test coverage seemed low for the complexity of the code under test, but if you're happy that it's sufficiently covered, that's good enough for me! 😄